tianshou/data/batch.py

```diff
@@ -534,21 +576,24 @@ def __len__(self) -> int:
         return min(r)
 
     @property
-    def size(self) -> int:
-        """Return self.size."""
+    def shape(self) -> List[int]:
+        """Return self.shape."""
         if len(self.__dict__.keys()) == 0:
-            return 0
+            return []
         else:
-            r = []
+            data_shape = []
             for v in self.__dict__.values():
-                if isinstance(v, Batch):
-                    r.append(v.size)
-                elif hasattr(v, '__len__') and (not isinstance(
-                        v, (np.ndarray, torch.Tensor)) or v.ndim > 0):
-                    r.append(len(v))
-                else:
-                    r.append(1)
-            return min(r) if len(r) > 0 else 0
+                try:
+                    if len(v.shape) > 0:
```

Comment:
Why to do an exception for scalar ? I don't think it is a good idea since it does not support slicing, and it is not consistent with `len` (`len(v) != v.shape[0]`)